app/services/blog_service.rb

```
@@ -0,0 +1,135 @@
+    all_posts.find(&:featured) # Assumes featured posts are also published
+  end
+
+  def self.recent_posts(limit = 5)
+    all_posts.take(limit)
+  end
+
+  private
+
+  # Memoized method to load and parse all .md files once per request/class load.
+  # Returns an array of PostData objects (both published and unpublished).
+  def self._load_all_parsed_posts
```

RuboCop (1.73)

[convention] Use 2 (not 0) spaces for indented_internal_methods indentation.

(Layout/IndentationWidth)